### ISSUE TITLE
Inconsistency between source and items types

### ISSUE THREAD

**Issue by ainvaltin** [ThomasCybulski/paper-chip#68]
`paper-chip-input-autocomplete`, currently the `source` must be an array of objects while `items` is always array of strings. 

It would be nice if the `source` would also accept an array of strings in which case the `items` would be strings. If the `source` is assigned an array of objects then `items` would also be an array of objects. When the `source` is objects then additional `item-label-path` property is used to determine which property of the objects is used for the chip label (defaults to `text` to preserve current behaviour).

When source is objects but there is also `item-value-path` property set then the `items` would contain values of those properties instead of objects.

**Comment by alexeygt** [comment]
I second the idea with objects used for item list. However there are few problems which should be carefully analyzed before implementing such request:

1. How would it work in case when "additional-items" mode is enabled. Create new object with new label, but empty value?
1. How would that work when "allow-duplicates" mode is enabled?

**Comment by ainvaltin** [comment]
> How would it work in case when "additional-items" mode is enabled. Create new object with new label, but empty value?

Yes, create new obj with property named by `item-label-path` (default "text"). There could be a callback which can be assigned by component's user to init the new object.

> How would that work when "allow-duplicates" mode is enabled?

Just create multiple objects, each containing same properties/values. It's not different from current situation where `items` contains same string multiple times.
